app/proguard-rules.pro

```diff
@@ -104,6 +104,12 @@
     <init>(android.app.Application);
 }
 
+####################################################################################################
+# Mozilla Application Services
+####################################################################################################
+
+-keep class mozilla.appservices.** { *; }
+
```

Comment:
I wonder if 

```
-keepclassmembers class * extends com.sun.jna.** {
    <fields>;
    <methods>;
}
```

would do the maximal thing you want— though **only** for APK size.

Reply:
I think we would get this for free via https://github.com/mozilla/application-services/blob/main/proguard-rules-consumer-jna.pro .

Is the idea to add this for all fields/methods and not just the public ones? I'm not sure if that would keep more than we use. 🤔

Reply:
Thinking about it more, this rule is the same one used in Fenix and since this change is blocking the A/A experiment, I'd like to keep the rule identical to what we have there.

Happy to re-visit this again. Maybe as part of https://github.com/mozilla-mobile/focus-android/issues/4922?